actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -295,13 +319,18 @@ def content_tag(name, content_or_options_with_block = nil, options = nil, escape
       # ==== Examples
       #   class_names("foo", "bar")
       #    # => "foo bar"
+      #   class_names("foo", "bar") + class_names("bar", "baz")
+      #    # => "foo bar baz"
+      #   "foo" + class_names("bar", { baz: true })
+      #    # => "foo bar baz"
       #   class_names({ foo: true, bar: false })
       #    # => "foo"
       #   class_names(nil, false, 123, "", "foo", { bar: true })
       #    # => "123 foo bar"
       def class_names(*args)
-        safe_join(build_tag_values(*args), " ")
+        TokenList.new(self, build_tag_values(*args))
       end
+      alias_method :token_list, :class_names
```

Comment:
@georgeclaghorn would an alias for `tokens` make sense here too?

Reply:
I wish we'd just remove class_names entirely and go with token_list as the name.

Reply:
Scratch that, since class_names hasn't been released yet, we should do that.

Reply:
@kaspth I've pushed up https://github.com/rails/rails/pull/40146/commits/ffd18391a5fe14ea71a378437584e7b6675967e3 to abandon the alias and rename the original.

Reply:
@kaspth I've reverted the re-name. Do you feel that the `dom_token_list` alias is still viable?